### PR TITLE
improve airdate function

### PR DESCRIPTION
**Pull request by mediaminister**
This includes:
- added support for `RunPlugin()`
- Live-to-VOD: if VRT doesn't provide a VOD stream, you still can rewatch programs from the last 24 hours from the archived livestream

### CODE REVIEW

File: resources/lib/streamservice.py [hunk 1]
```diff
@@ -216,6 +216,10 @@ def get_stream(self, video, roaming=False, api_data=None):
             # Get stream manifest url
             manifest_url = next(stream.get('url') for stream in stream_json.get('targetUrls') if stream.get('type') == protocol)
 
+            # External virtual subclip support (live2vod airdate feature)
+            if video.get('start_date') and video.get('end_date'):
+                manifest_url += '?t=' + video.get('start_date') + '-' + video.get('end_date')
```
Comment (dagwieers) [inline resources/lib/streamservice.py]: Nice !

File: resources/lib/streamservice.py [hunk 2]
```diff
@@ -216,6 +216,10 @@ def get_stream(self, video, roaming=False, api_data=None):
...
+
             # Fix virtual subclip
             duration = timedelta(milliseconds=stream_json.get('duration'))
```
Comment (dagwieers) [inline resources/lib/streamservice.py]: When using `get()` I always think about, what would happen if `duration` is not found?
Would milliseconds=None work, or do we need to provide a working default?
```
$ python -c 'from datetime import timedelta; print(timedelta(milliseconds=None))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: unsupported type for timedelta milliseconds component: NoneType
```

Comment (dagwieers) [reply resources/lib/streamservice.py]: Same for the other `get()` calls obviously.

File: resources/lib/kodiwrapper.py [hunk 3]
```diff
@@ -229,7 +229,11 @@ def play(self, video):
             play_item.setSubtitles([video.subtitle_url])
 
         self.log_notice('Play: %s' % unquote(video.stream_url), 'Info')
-        xbmcplugin.setResolvedUrl(self._handle, bool(video.stream_url), listitem=play_item)
+        if self._handle != -1:
```
[the remark this answers is not in the document]
Comment (mediaminister) [reply resources/lib/kodiwrapper.py]: Done.

File: resources/lib/streamservice.py (hunk 1 above)
Comment (mediaminister) [reply resources/lib/streamservice.py]: Unfortunately, the livestream is only archived for 24 hours. I edited the comment accordingly.

**Pull request merged**